src/humidity.c

```diff
@@ -2,34 +2,90 @@
 #include <avr/delay.h>
 #include <hih8120.h>
 
-int measureHumidity() {
-    if ( HIH8120_OK == hih8120_initialise() )
+static TaskHandle_t _HumiditySensorTaskHandle;
+
+//private fields
+static SemaphoreHandle_t _xPrintfSemaphore;
+static EventGroupHandle_t _eventGroupHandleMeasure;
+static EventGroupHandle_t _eventGroupHandleNewData;
+static float _lastMeasurementHumidity;
+
+// Delay constants
+#define DELAY_WAKEUP 15
+#define DELAY_MEASUREMENT 20
+
+/**
+ * Initializes the humidity sensor and checks if the initialization was successful.
+ * Prints a message indicating the status of the driver.
+ */
+void initialiseHumidity()
+{
+    // Call the humidity sensor initialization function
+    hih8120_driverReturnCode_t returncode = hih8120_initialise();
+
+    // Check if the initialization was successful
+    if (HIH8120_OK == returncode)
+    {
+        printf("Humidity driver has been initialized\n");
+    }
+    else
     {
-        // Driver initialised OK
-        // Always check what hih8120_initialise() returns
+        printf("Driver not working\n");
     }
+}
 
+
+
+void measureHumidity()
+{
     float humidity = 0.0;
-    //float temperature = 0.0;
+    // Wait until the measure event is triggered
+    xEventGroupWaitBits(_eventGroupHandleMeasure,
+                        pdTRUE,
+                        pdTRUE,
+                        portMAX_DELAY);
```

Comment:
`src/humidity.c:43:5: error: too few arguments to function 'xEventGroupWaitBits'
     xEventGroupWaitBits(_eventGroupHandleMeasure`

Also too few arguments...